### ISSUE TITLE
Unfortunately, your game has stopped.

### ISSUE THREAD

**Issue by tommyzat** [playgameservices/play-games-plugin-for-unity#1513]
Hello!
Google play games services works fine for me, achievements work etc.
Whenever I exit my game, after ~3 seconds, I get the error that my game crashes. This only happens with the google services implemented.
Using Unity 5.5.0p1 and latest google services version

**Comment by tommyzat** [comment]
This really is an urgent issue, please help

**Comment by alexhofer** [comment]
Do you happen to have a log? A stack trace grabbed from logcat or the android debugger would help figure out whats throwing the error.

**Comment by tommyzat** [comment]
@teemukorh could you please provide a logcat? I accidentally disabled it on my phone.

**Comment by FrankDaze** [comment]
I have the same problem. But it's hard to get a log, because if I export my game in Dev Mode, the game is running fine. If I disable Dev Mode, it's crashing after a few seconds.

**Comment by alexhofer** [comment]
Since that error sounds like its from the OS itself, even without dev mode on the catlog might still be useful. At least you can see where the error is getting thrown. 

Additionally have you tried clearing the cache for the app, and fully uninstalling it? There could be some issue with some data corruption or old data being left between builds.

**Comment by FrankDaze** [comment]
ok, I'm out. I found my problem, it was not the google play games plugin. The problem came from a particle effect.

**Comment by teemukorh** [comment]
Here is the log from my device. Seems to crash in native code.
[logcat.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/652434/logcat.txt)

**Comment by teemukorh** [comment]
I'm also using Unity 5.5.0p1, which seems to be quite unstable release.

**Comment by daegney** [comment]
This is the log from my device after calling Application.Quit(). (Unity 5.5.0p1, #1500)

D/dalvikvm: Unregistering JNI method Lorg/fmod/FMODAudioDevice;.fmodProcess:IL
D/dalvikvm: Unregistering JNI method Lorg/fmod/FMODAudioDevice;.fmodGetInfo:II
D/dalvikvm: Unregistering JNI method Lorg/fmod/FMODAudioDevice;.fmodProcessMicData:ILI
W/Unity: Timeout while trying to pause the Unity Engine.
A/libc: Fatal signal 11 (SIGSEGV) at 0x00000000 (code=1), thread 12003

This only happens with PlayGamesPlatform.Activate().

**Comment by tommyzat** [comment]
This issue was fixed for me by publishing the game to Google Play for some reason. I'm leaving this open for others with the same issue

**Comment by teemukorh** [comment]
The crash is there. The os just doesn't show the dialog for games installed from Google Play.

**Comment by 5argon** [comment]
Confirmed the same problem. Development Build is fine, while non-dev build will error on initialize GPGS. Does not happen on Nexus 5, but happen on Zenfone 2 Laser. Both tested by directly installing the APK.

**Comment by claywilkinson** [comment]
Can you share the log of the crash?

**Comment by leonard4** [comment]
Seeing the same issues, Unity 5.5.0p2 with the newest 10.0.1 GPG files, seeing an ANR about 3-5 seconds after issuing an Application.Quit(). I pulled all of the GPG files and references and the app closes fine now. Also on a side note using 5.5.0p2 with 9.80 worked fine, so something in the newer 10 version that has some issue.

**Comment by leonard4** [comment]
Just wanted to add some info, upgraded to 5.5.0p3 and built using Mono2x as the runtime, same crash. Tried building to IL2CPP since it was fixed in 5.5.0p3 and the game does not crash on Quit, not sure if that helps point in the right direction.

**Comment by claywilkinson** [comment]
Thanks - that is an interesting data point.

**Comment by ambs** [comment]
same issue, IL2CPP solves as well.

**Comment by gyazbek** [comment]
@leonard4 Just tried switching to IL2CPP with Unity 5.5.0f3 and still running into the same crash.

**Comment by fyn777** [comment]
same issue. In unity 5.3.2f1 worked good. But after updating to 5.5.1f1 get this issue. Any solutions?

**Comment by gyazbek** [comment]
@fyn777 I did a small workaround until this is fixed. Send the task activity to the back instead of quitting. Not the most ideal since it isn't really quitting, and you depend on the system to manage the rest, but it works in a pinch while I develop. 

It would be something like:
```
#if UNITY_ANDROID
new AndroidJavaClass("com.unity3d.player.UnityPlayer").GetStatic<AndroidJavaObject>("currentActivity").Call<bool>("moveTaskToBack" , true); 
#endif
```

**Comment by trtuanpham** [comment]
same issue, version 0.9.36

**Comment by trtuanpham** [comment]
@gyazbek thanks for your help

**Comment by asadrana91** [comment]
have the same issue, anyone found any fix?@gyazbek your solution minimizes the app or closes the unity player but game doesnt exit.

**Comment by gyazbek** [comment]
@asadrana91, Correct. As I mentioned, 'Send the task activity to the back instead of quitting.' This allows the system to handle when it should free up resources, unless your application is specifically marked as a background service.

**Comment by asadrana91** [comment]
instead of using Application.Quit() ,try us System.Diagnostics.Process.GetCurrentProcess().Kill(); , resolved the issue for me.

**Comment by stevencodeuk** [comment]
@asadrana91 I've just tried

`System.Diagnostics.Process.GetCurrentProcess().Kill();`

and the game quits without crashing. However, my device shows my game still as an active app, albeit it without any screenshot attached to. Might be worth checking your end to ensure quit is working.

**Comment by leonard4** [comment]
Just wanted to add that this still happens as of Unity 5.5.1p4. IL2CPP appears to be broken in 5.5.1 so building in Mono is the only option, but the games crash on app.quit(), makes 5.5.1 unusable atm with the GPG plugin. :(

**Comment by mj321** [comment]
Same problem here, switching to IL2CPP fixed it for me. Thanks for the tip.

Unity 5.5.0.f3.

**Comment by Firemaw** [comment]
Downloading the NDK r10e with Unity 5.5.3f1 and setting Android Player Settings Scripting Backend = IL2CPP fixed the issue on my Android 4.1 and 6.1 devices. However, build times are about 3-4 times longer than Mono2x.

**Comment by claywilkinson** [comment]
I believe the latest version of the plugin 0.9.37 addresses the Application.Quit causing a crash issue.

As far as long build times for IL2CPP, is it feasible to do most of the development using mono, then switch to IL2CPP towards the end of the release cycle?  Are you seeing a lot of differences between mono and IL2CPP that require you to use it for the full development process?

**Comment by Firemaw** [comment]
Yes, I had been using Mono2x for dev, then switching to IL2CPP for final build and test. I will try out the new version 0.9.37. Thanks Clay!

**Comment by Firemaw** [comment]
I have updated to 0.9.37 and can confirm the Application.Quit error is fixed with Mono2x on my Android 4.4 and 6.1 devices. Thanks Clay!

**Comment by thiagolr** [comment]
This is still happening on version 0.9.37! The System.Diagnostics.Process.GetCurrentProcess().Kill(); workaround works.

**Comment by olehkuznetsov** [comment]
Should be fixed already in v0.10.06. Feel free to reopen if it reproduces with latest plugin version.
